### ISSUE TITLE
pkg-config generator hardcodes folder paths

### ISSUE THREAD

**Issue by jmoguillansky-gpsw** [mesonbuild/meson#8169]
**Describe the bug**
When using pkg-config generator the paths "include" and "lib" are hardcoded.  
Can we support more flexible options:
allow the user the specify the include directory, library directory, binary directory, etc.

**To Reproduce**
pkg.generate(
libxyz,
name: 'libxyz',
)

**Expected behavior**
Can we expose additional params like libdir, includedir, etc?

**system parameters**
Windows 10, Python 3.8.0, meson version 0.55.3

**Comment by eli-schwartz** [comment]
Whats the goal here? The purpose of the generator is to create pkg-config files referring to the installed library, right? So, you want it to respect pkg-config built-in variables like `${prefix} ${exec_prefix} ${includedir} ${libdir}` but hardcode it to ultimately resolve to the installed library. Users doing complex things with their toolchain can modify how this works at runtime using `pkg-config --define-variable`, but without such options it should always refer to the installed library.

Is there something wrong with the generator causing it to *not* refer to the installed library???

**Comment by jmoguillansky-gpsw** [comment]
So here's more context regarding my specific issue:
I'm building a native library, but then installing it to a "python virtual environment" on Windows.  
When I create the python virtual environment on windows, it uses the folder structure "Scripts", "Include", "Lib", etc (on Linux it uses "bin", "include", "lib").  

When I generate the pkg-config file with meson on Windows using the pkg-config generator, it sets includedir as ${prefix}/include, libdir as ${prefix}/lib.  
Can I customize these paths, e.g. set includedir to ${prefix}/Include (capital I), libdir to ${prefix}/Lib (capital L), etc.  
In other words can I have more control over the generated pkg-config files and the variables?

**Comment by eli-schwartz** [comment]
Maybe:

```
meson setup builddir --prefix=C:\\Path\\To\\Virtualenv --libdir=Lib --includedir=Include --bindir=Scripts
```

Then `meson install` will install the library to `C:\\Path\\To\\Virtualenv\\Lib`, the headers to `C:\\Path\\To\\Virtualenv\\Include`, the programs to `C:\\Path\\To\\Virtualenv\\Scripts`... and the pkg-config generator should encode these locations too.

**Comment by jmoguillansky-gpsw** [comment]
Thanks that fixed the issue!

**Comment by eli-schwartz** [comment]
Great, happy to help!
